### DISCUSSION TITLE
fastify-jwt

### DISCUSSION THREAD

**Discussion by yuanye2005king** [L2jLiga/fastify-decorators#1017] [Q&A] [answered]
fastify-decorators is a great plugin, and I have already applied it in our project. Now I have a question, how can I apply fastify-jwt to the controller to implement token verification for certain controllers? Thank you very much.

**Comment by L2jLiga** [comment, marked as answer]
Hi there, according to [fastify-jwt docs] you can verify token inside onRequest hook. On fastify-decorators side exists [Hook] decorator which can be used to define hook in controller. Each controller registers in fastify as plugin and hence each controller has [own scope], so hooks defined in controller will work only in controller where you define them.

Snippet:

```typescript
@Controller('/protected')
class ProtectedController {
  @Hook('onRequest')
  async (request, reply) => {
    try {
      await request.jwtVerify()
    } catch (err) {
      reply.send(err)
    }
  }
}
```

[fastify-jwt docs]: https://github.com/fastify/fastify-jwt?tab=readme-ov-file#usage
[Hooks]: https://github.com/L2jLiga/fastify-decorators/blob/v3/docs/Controllers.md#creating-hooks
[own scope]: https://github.com/fastify/fastify/blob/4.x/docs/Reference/Encapsulation.md
